### PR TITLE
Add DELETE option to github-add-comment

### PR DESCRIPTION
**Pull request by abayer**
# Changes

This will be used as part of https://github.com/tektoncd/plumbing/issues/483 in Tekton's own CI setup, and I think it could be generally useful for replicating behavior such as Prow's job failure comments. Those comments are deleted once the job completes next, with a new comment created for subsequent failures, so that the latest result is always towards the end of the comment list.

/kind feature

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [x] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [x] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs] https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests] https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e] https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor] https://github.com/tektoncd/community/blob/main/standards.md
[commit] https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

**Comment by abayer** [comment]
...you may notice that I don't know python at all well, so I'm not sure what's breaking. =)

File: task/github-add-comment/0.8/github-add-comment.yaml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+        split_url = urllib.parse.urlparse(
+            "$(params.REQUEST_URL)").path.split("/")
+
+        # This will convert https://github.com/foo/bar/pull/202 to
+        # api url path /repos/foo/issues/
+        api_url = "{base}/repos/{package}/issues/{id}".format(
+          base="$(params.API_PATH_PREFIX)", package="/".join(split_url[1:3]), id=split_url[-1])
+
+        commentParamValue = """$(params.COMMENT_OR_FILE)"""
```
Comment (dibyom) [inline task/github-add-comment/0.8/github-add-comment.yaml:126]: If the workspace is not bound, will we just end up making a comment with the filename?

File: task/github-add-comment/0.8/github-add-comment.yaml [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        matching_comment = ""
+        if "$(params.REPLACE)" == "true" or "$(params.DELETE)" == "true":
+          if not "$(params.COMMENT_TAG)":
+            print("REPLACE or DELETE requested but no COMMENT_TAG specified")
+            sys.exit(1)
+          r = conn.request(
+            "GET",
+            api_url + "/comments",
+            headers={
+                "User-Agent": "TektonCD, the peaceful cat",
```
Comment (dibyom) [inline task/github-add-comment/0.8/github-add-comment.yaml:156]: XD

**Review by dibyom — APPROVED** [review]

**Comment by abayer** [comment]
/hold

So I don't trust my Python skills enough - I'm working on a custom task specific for the Tekton project's own use cases. =)

**Pull request closed without merging**